### PR TITLE
Implements search on repositroy base

### LINKED ISSUE #43: Search among Fortnox resources
Many Fortnox resource attributes is "searchable". It would be great to implement this functionality in our gem.

More about [Fortnox search](http://developer.fortnox.se/documentation/general/parameters/#Search).

### PR DESCRIPTION
**Pull request by d-Pixie**
Fixes #43

### CODE REVIEW

File: spec/fortnox/api/repositories/customer_spec.rb [hunk 1]
```diff
@@ -4,13 +4,16 @@
 require 'fortnox/api/repositories/examples/all'
 require 'fortnox/api/repositories/examples/find'
 require 'fortnox/api/repositories/examples/save'
+require 'fortnox/api/repositories/examples/search'
 
-describe Fortnox::API::Repository::Customer do
+describe Fortnox::API::Repository::Customer, order: :defined, integration: true do
```
Comment (ehannes) [inline spec/fortnox/api/repositories/customer_spec.rb]: Great with the `order: :defined`!

Comment (ehannes) [inline spec/fortnox/api/repositories/customer_spec.rb]: And the `integration: true`!

File: spec/fortnox/api/repositories/examples/search.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+      context "with one match" do
+        subject do
+          VCR.use_cassette( "#{vcr_dir}/search_by_name" ) do
+            repository.search( attribute_hash_key_name => value )
+          end
+        end
+
+        it{ is_expected.to be_instance_of( Array) }
+        it{ is_expected.to have(1).entries }
```
Comment (ehannes) [inline spec/fortnox/api/repositories/examples/search.rb]: Should we verify that the returned entry has the attribute that we searched for?

**Comment by ehannes** [comment]
Looking great!

**Comment by ehannes** [comment]
How do we add filters and searchable attributes to the repositories?

File: spec/fortnox/api/repositories/examples/search.rb (hunk 2 above)
Comment (d-Pixie) [inline spec/fortnox/api/repositories/examples/search.rb]: No, that logic is not ours, so there is no point in verifying it.

**Comment by d-Pixie** [comment]
Right now we allow any for both search and filter. I'll add whitelisting next.

**Pull request merged**